time stamp
uint32 states
float32[] matrix

